src/realm/query_expression.hpp

```diff
@@ -3678,27 +3678,21 @@ class Columns : public ObjPropertyExpr<T> {
             // Not a Link column
             size_t colsize = leaf->size();
 
+            size_t rows = std::min(colsize - index, ValueBase::chunk_size);
+
             // Now load `ValueBase::chunk_size` rows from from the leaf into m_storage.
             if constexpr (std::is_same_v<U, int64_t>) {
                 // If it's an integer leaf, then it contains the method get_chunk() which copies
-                // these values in a super fast way (only feasible if more than chunk_size in column)
```

Comment:
"only feasible if more than chunk_size in column" was actually just always incorrect and this always could have used get_chunk().